### ISSUE TITLE
Compatibility to integrate Hotjar

### ISSUE THREAD

**Issue by PilarDCV** [ampproject/amphtml#27545]
Hello, it would be very important to be able to integrate Hotjar scripts (https://www.hotjar.com/) on my website (wordpress + custom theme amp). I have researched the documentation for amp components and the list of supported vendors but can't find any option to do so.

I had high expectations with the arrival of amp-script, but it is not working to implement tracking scripts.

Can any of the current components be used to insert tracking scripts? Which one?

If this is not possible using a current component, do you consider the development of the feature viable? What is the usual deadline for new features?

Thank you!

**Comment by caroqliu** [comment]
Have you taken a look at [amp-analytics ](https://amp.dev/documentation/guides-and-tutorials/optimize-and-measure/configure-analytics/analytics-vendors/) to see if it meets your needs? @ampproject/wg-analytics

**Comment by rrjanbiah** [comment]
@caroqliu FWIW, Hotjar is one of the top players in this Analytics segment https://trends.builtwith.com/analytics But, it is not available in https://github.com/ampproject/amphtml/tree/master/extensions/amp-analytics/0.1/vendors

**Comment by micajuine-ho** [comment]
Hi @rrjanbiah! Thanks for your issue. Hotjar can be integrated into amp-analytics using [this guide](https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/integrating-analytics.md). However, it requires help from the vendor in question to make sure that the requests from AMP will be properly captured.

Is think something you would like to contribute to the repo?

**Comment by rrjanbiah** [comment]
@micajuine-ho I have reached out to the Hotjar team. Based on their reply, will update

**Comment by rrjanbiah** [comment]
@micajuine-ho While I waited for them to reply, I quickly checked. It looks like current `amp-analytics` architecture wouldn't allow click tracking, heatmap, etc. In case, if that's already possible, can you please share the vendor that's already doing that?

**Comment by rrjanbiah** [comment]
From Hotjar support:

> Unfortunately, AMP does not support third-party scripts (like ours) on their pages outside of iframes. Since [iframes are sandboxed](https://amp.dev/documentation/guides-and-tutorials/develop/media_iframes_3p/third_party_components/) away from the rest of the site, it means Hotjar cannot access page information on your AMP pages. Unfortunately, unless Google changes how AMP works here, Hotjar will be unable to work.

**Comment by micajuine-ho** [comment]
@rrjanbiah There are a lot of vendors that use the "click" trigger. What are heatmaps?

**Comment by rrjanbiah** [comment]
@micajuine-ho Analytics is more than clicks...  many providers offer playback of mouse moves aka session recordings.

For example, here's the one from Hotjar:

Animated GIF https://assets.hotjar.com/cs-team/knowlege_base/recordings/player_playback_viewer_demo.gif

**Comment by micajuine-ho** [comment]
Hi @rrjanbiah, amp-analytics does not offer session recordings as a trackable event. [Here](https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/amp-analytics.md#triggers) is a list of all the triggers that we offer. 

Is session recordings something that you need?

**Comment by rrjanbiah** [comment]
@micajuine-ho Yes, that's an important requirement for any analytics company. Is there anything you can do to push it?

**Comment by micajuine-ho** [comment]
Is session recordings the only new feature that is necessary for your company? Or are there others?

**Comment by rrjanbiah** [comment]
@micajuine-ho Even GA had heatmap feature once, but they removed it. I think, clickmap, scrollmap, session recording (hover and keyboard events) are important. Noticed that hover is longtime pending #15114

Also, an example analytics provider https://www.inspectlet.io/includes/assets/session-recording-demo.mp4 with session recording.

**Comment by micajuine-ho** [comment]
@rrjanbiah We can take on this work in the future as a part of supporting custom browser events, such as hovering, keyboard events, and clicked.

**Comment by luiznasciment0** [comment]
Hello! Any updates on this? We're needing heatmap (hotjar) here too :(

We thought about disable/invalidate AMP, then try to use Hotjar, is it possible?

**Comment by micajuine-ho** [comment]
Hi, thank you for the bump. No updates as of yet, but we can start to plan the prioritization of the Tracking of Custom Browser Events.

**Comment by luiznasciment0** [comment]
It would be amazing. This kind of tool is essential for metrics and UX/UI

**Comment by luiznasciment0** [comment]
> I'm sorry, what do you mean by this?

I mean like, insert hotjar script and install it manually, then AMP would be invalid because of these third party scripts, but my hotjar (heatmap) would work, wouldn't? 

I made a logic that I could enable/disable hotjar, and AMP would be invalid/valid in consequence

**Comment by micajuine-ho** [comment]
> I mean like, insert hotjar script and install it manually, then AMP would be invalid because of these third party scripts, but my hotjar (heatmap) would work, wouldn't?

Ah I see what you mean. Yes you could install Hotjar script to run the Hotjar heatmap, but then it would be invalid AMP. 

With the Tracking of Custom Browser Events that I mentioned above, once we get that done, we can integrate Hotjar as a first-party analytics vendor and then you use their analytics through `<amp-analytics>` and have valid AMP.

**Comment by luiznasciment0** [comment]
> > I mean like, insert hotjar script and install it manually, then AMP would be invalid because of these third party scripts, but my hotjar (heatmap) would work, wouldn't?
> 
> Ah I see what you mean. Yes you could install Hotjar script to run the Hotjar heatmap, but then it would be invalid AMP.
> 
> With the Tracking of Custom Browser Events that I mentioned above, once we get that done, we can integrate Hotjar as a first-party analytics vendor and then you use their analytics through `<amp-analytics>` and have valid AMP.

This is going to be AWESOME. Thanks a lot for the attention and good luck on this project!

**Comment by J-uliane** [comment]
Hey @micajuine-ho ,

Looking at integrating with Hotjar too. Has there been any progress on your last message from Dec 11, 2020?

Thanks

**Comment by dmanek** [comment]
We're working on supporting custom browser events with initial support for `change` and `blur` events. Stay tuned!

**Comment by Gott50** [comment]
> Hi, thank you for the bump. No updates as of yet, but we can start to plan the prioritization of the Tracking of Custom Browser Events. We can track the project here #17878
> 
> Once this project is in, then we can follow up with the Hotjar team and see if they can integrate into `amp-analytics`.

@micajuine-ho

#17878 is closed
Has there been any progress here? ^^‘

Thank You :)

**Comment by rrjanbiah** [comment]
@Gott50 I think, @micajuine-ho is not working with Google now. Based on above comments, @dmanek is now working on it. But, not sure about the ETA
